### PR TITLE
Details Component Basic Implementation

### PR DESCRIPTION
**Pull request by korenkovv**
This is a basic implementation of the details component. It doesn't assume any client-server communication yet.
Graphs are only available now.
Layers and KPIs should be added later on.

Graph Preview added.

### CODE REVIEW

File: WebClient/doc/core/modelCollection/modelCollection.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+| Property | Type | Description |
+|---|---|---|
+| models | Array&lt;object&gt; | Array containing model objects. Changing this array will not have an effect on the actual property but changing the model objects will actually change them.  |
+
+#### Methods
+
+| Method | Returns | Description |
+|---|---|---|
+| set(Array&lt;object&gt; models) | void | Sets the models to the collection. Previously stored models will be removed from this collection. `change` event is emitted. If models are removed from the collection then `remove` event is emitted. |
```
Comment (cornelissenja) [inline WebClient/doc/core/modelCollection/modelCollection.md:26]: Is the change event emitted per model removed?
Make more specific..

File: WebClient/doc/core/modelCollection/modelCollection.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+| add(Array&lt;object&gt; models \| object model) | void | Adds a sigle model or an array of models to the collection. Previously stored models will be removed from this array. `change` event is emitted. |
+| remove(Array&lt;object&gt; models \| object model) | void | Removes a sigle model or an array of models from the collection. `change` event is emitted. If models are removed from the collection then `remove` event is emitted. |
```
Comment (cornelissenja) [inline WebClient/doc/core/modelCollection/modelCollection.md]: type sigle should be single? 2 times..

File: WebClient/doc/core/modelCollection/modelCollection.md (hunk 1 above)
Comment (cornelissenja) [reply WebClient/doc/core/modelCollection/modelCollection.md]: Make more specific..

File: WebClient/doc/core/modelCollection/modelCollection.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+| contains(object model) | void | returns `true` if the model is stored in the collection. Otherwise returns `false`. |
+| getById(string modelId) | object | Returns model object with the given id or returns `null` if such model is missing in the collection. |
+| empty() | boolean | returns `true` if the collection is empty and doesn't contain any models. Otherwise returns `false`. |
+
+#### Events
+
+| Property | Data | Description |
+|---|---|---|
+| change | { models:Array &lt;object&gt; } | Emitted whenever models collection change duw to models added or removed. |
```
Comment (cornelissenja) [inline WebClient/doc/core/modelCollection/modelCollection.md]: typo: duw=due i think..

File: WebClient/doc/core/modelCollection/modelCollection.md (hunk 1 above)
Comment (korenkovv) [reply WebClient/doc/core/modelCollection/modelCollection.md:26]: One event per function call. This means if multiple models are removed within a single `set()` call then one `change` event and one `remove` event will be emitted. `Change` event will provide the new state of the collection while `remove` event will provide the list of removed models.

Comment (korenkovv) [reply WebClient/doc/core/modelCollection/modelCollection.md:26]: I'll extend the description to be more precise

File: WebClient/doc/core/view/view.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+| Method | Returns | Description |
+|---|---|---|
+| _notifyFocus() | This method sends the `focus` event to *View* subscribers. |
+
+#### Events <a name="events"></a>
+
+| Event | Data | Description
+|---|---|---|
+| remove | { view: *View* } | Emitted when the view is removed via `remove()` method. |
+| focus | { view: *View* } | Can be emitted from child classes when view is focused focus. |
```
Comment (cornelissenja) [inline WebClient/doc/core/view/view.md]: ..focused focus. Now I'm really confused..
This should be clarified.

Comment (korenkovv) [reply WebClient/doc/core/view/view.md]: Will remove extra `focus` word :)

File: WebClient/doc/core/window/windowView/windowView.md [hunk 5]
```diff
@@ -53,27 +51,29 @@ Options object that should be passed to *WindowView* constructor.
...
+See also [View](../../view/view.md#methods) methods.
+
+#### Extension methods <a name="extensionMethods"></a>
 Every window should extend from *WindowView* and may re-implement (override) the following methods.
 
 | Method | Returns | Description |
 |---|---|---|
-| onRender(Node viewport) | void | Method is called when window is being rendered. At he time when this method is called all window's DOM nodes are already created so it is safe to refer them from within this method. |
-| onRemove() | void | Method is called when window is being removed. |
+| onRenderWindow(Node viewport) | void | Method is called when window is being rendered. At he time when this method is called all window's DOM nodes are already created so it is safe to refer them from within this method. |
```
Comment (cornelissenja) [inline WebClient/doc/core/window/windowView/windowView.md]: typo: "at he time when this" should be "at the time this" ?

File: WebClient/src/components/details/detailsItemView.js [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+        var previewViewport = d3.select(this._rootElement).select('.preview-viewport').node();
+
+        this._preview = this._previewCreator(previewViewport);
+
+        return this._rootElement;
+    },
+
+    onRemove: function () {
+        this._model.off('title', this._updateTitle, this);
+        this._model.off('slected', this._updateSelection, this);
```
Comment (cornelissenja) [inline WebClient/src/components/details/detailsItemView.js]: typo: slected=selected

File: WebClient/src/components/details/itemsContainer.js [hunk 7]
```diff
@@ -0,0 +1,176 @@
+/**
+ * ItemsContainer is a generic container to store model and their views.
```
Comment (cornelissenja) [inline WebClient/src/components/details/itemsContainer.js]: Maybe add a reference that ItemsContainer contains Detail item that contain model and view

File: WebClient/src/components/details/itemsContainer.js [hunk 8]
```diff
@@ -0,0 +1,176 @@
+/**
```
Comment (cornelissenja) [inline WebClient/src/components/details/itemsContainer.js]: Maybe rename file to be more consistent with other file names

File: WebClient/src/components/graph/graphViewControllerFactory.js [hunk 9]
```diff
@@ -16,12 +16,11 @@ var GraphViewControllerFactory = L.Evented.extend({
         };
     },
 
-    create: function (graphModel, windowManager, graphViewOptions) {
+    create: function (graphModel, graphViewOptions) {
         var ViewController = this._viewControllerMap[graphModel.type] || GraphViewController;
```
Comment (cornelissenja) [inline WebClient/src/components/graph/graphViewControllerFactory.js:22]: Maybe a good idea to add a comment here why the generic GraphViewController is or-ed with the look-upped type

**Review by cornelissenja — APPROVED** [review]
🥇

**Pull request merged**